### PR TITLE
Update Stencil to latest version and setup testing infrastructure.

### PR DESCRIPTION
**Pull request by abuiles**
This pull request updates Stencil to the latest version and setups testing infrastructure for unit and e2e testing.

It also adds a very simple example showing how to test some of the current helpers included in the project and explains how to use the testing framework.

### CODE REVIEW

File: .npmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+package-lock=false
```
[the remark this answers is not in the document]
Comment (JakeUrban) [reply .npmrc:1]: Oh never mind, I see you just swapped `package-lock.json` for `yarn.lock`

**Review by JakeUrban — APPROVED** [review]
This is great, thanks @abuiles. I left a couple comments but nothing blocking.

File: .vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{
```
[the remark this answers is not in the document]
Comment (abuiles) [reply .vscode/launch.json:1]: hey! yeah, that's the working directory -- it does not require developers to have VSCode, but if you do you get this for free, if you don't then that's okay, you can still do `yarn test` or `yarn watch`

[the remark this answers is not in the document]
Comment (abuiles) [reply .vscode/launch.json:1]: For reference, we do something similar in the go monorepo https://github.com/stellar/go/tree/master/.vscode -- again, there is no downside if you don't use vscode :)

**Pull request merged**